products-service/src/libs/api-gateway.ts

```diff
@@ -0,0 +1,34 @@
+import type { APIGatewayProxyEvent, APIGatewayProxyResult, Handler } from "aws-lambda"
+import type { FromSchema } from "json-schema-to-ts";
+
+type ValidatedAPIGatewayProxyEvent<S> = Omit<APIGatewayProxyEvent, 'body'> & { body: FromSchema<S> }
+export type ValidatedEventAPIGatewayProxyEvent<S> = Handler<ValidatedAPIGatewayProxyEvent<S>, APIGatewayProxyResult>
+
+type TResponseProduct = {
+  description: string;
+  id: string;
+  price: number;
+  title: string;
+}
+
+export const formatJSONResponse = (response: TResponseProduct[]) => {
+  return {
+    statusCode: 200,
+    headers: {
+      'Access-Control-Allow-Origin': '*',
+      'Access-Control-Allow-Credentials': true,
+    },
+    body: JSON.stringify(response)
+  }
+}
+
+export const formatNotFoundError = (textError: string) => {
+  return {
+    statusCode: 404,
+    headers: {
+      'Access-Control-Allow-Origin': '*',
+      'Access-Control-Allow-Credentials': true,
+    },
+    body: textError
```

Comment:
All your requests return "content-type": "application/json" but you return only text. So when I go to product/{productId} api I get json parse error. You should return JSON.stringify({ message:... }) or change content type in headers

Reply:
fixed